src/TraceEvent/Stacks/Linux/LinuxPerfScriptStackSource.cs

```diff
@@ -291,17 +291,27 @@ public LinuxPerfScriptStackSource(string path, bool doThreadTime)
 
         public double TotalBlockedTime { get; set; }
 
+        #region private
+        internal /*protected*/ GrowableArray<LinuxPerfScriptStackSourceSample> m_LinuxPerfScriptSamples;
+        #endregion
+
+        public LinuxPerfScriptStackSourceSample GetLinuxPerfScriptSampleByIndex(StackSourceSampleIndex sampleIndex)
+        {
+            return m_LinuxPerfScriptSamples[(int)sampleIndex];
```

Comment:
One other follow-up: I'd like to avoid this extra GrowableArray `m_LinuxPerfScriptSamples`.  What about implementing this API in terms of `GetSampleByIndex` and just cast the output?  Any concerns with this?

```suggestion
            return (LinuxPerfScriptStackSourceSample)GetSampleByIndex(sampleIndex);
```

Reply:
yes - I thought about this and tried this. It doesn't seem possible. You can downcast but not upcast. 

public StackSourceSample AddSample(StackSourceSample sample) is used to create a "real" sample. It was confusing to me as well how the existing code behaves

You cannot upcast StackSourceSample to LinuxPerfScriptStackSourceSample, and obviously StackSourceSample doesn't hold CpuNumber. So you need to add it and keep track of LinuxPerfScriptStackSourceSample somewhere because StackSourceSample is incapable of doing it

Reply:
You're right.  It's because `CopyStackSource.AddSample` makes a copy of the sample.